### ISSUE TITLE
Feature: Query CL capability 

### ISSUE THREAD

**Issue by yorickdowne** [ssvlabs/ssv#1533]
**Is your feature request related to a problem? Please describe.**

Consensus Layer clients might not implement new REST API endpoints at the same time. For example, Nimbus didn’t implement [/eth/v2/beacon/blinded_block](https://github.com/ethereum/beacon-APIs/pull/317) until July 2024.

As a result, block production with SSV and a pure Nimbus cluster failed.

**Describe the solution you'd like**

Query the available endpoints / CL capabilities on startup. Vouch code can be used as inspiration, as it does that.

Then either use the capabilities as discovered or, if a missing capability is mandatory, fail with a clear error message.

**Describe alternatives you've considered**

Robust QA to make sure SSV works with all CLs. The trouble is that this is hard to handle when new CLs come on the market, like Grandine and Lambda. Querying capabilities doesn’t replace QA, but it does provide a baseline that makes the QA job easier.

**Comment by y0sher** [comment]
@yorickdowne , thanks for opening this issue.
It's indeed a good idea, it will be released in next version!

**Comment by moshe-blox** [comment]
hey @yorickdowne 

i think this is a good idea, however not sure how we can confirm support for a route such as SubmitBlindedBeaconBlock which requires POSTing an actual block?

we can send an empty POST request and fail if we get 404, but i don't know if it's safe to trust that Beacon nodes won't return 404 just because there was no block in the request (rather than them not supporting this route altogether)

**Comment by yorickdowne** [comment]
Yeah that’s a good point. You could supplement with better error reporting. It took too long to know that the missed blocks were due to a missing endpoint. Logging more of the response on failure … code and message really … may have pinpointed that sooner

**Comment by Tom-ssvlabs** [comment]
We improved the error logging in the Ethereum clients.
